File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
T = int(input())
for test_case in range(1, T + 1):
    N, M = map(int, input().split())
    arr = [[] for _ in range(N)]
    for i in range(N):
        arr[i] = list(map(int, input().split()))
    answer = 0
    # 끝 점은 N-M+1 까지 가야 마지막까지 찾을 수 있다.
    answer_list = []
    for r in range(N-M+1):
        for c in range(N-M+1):
            answer = 0
            for i in range(M):
                for j in range(M):
                    answer += arr[r+i][c+j]
            answer_list.append(answer)

    print(f'#{test_case} {max(answer_list)}')

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.Scanner;
import java.io.FileInputStream;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution
{
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();

		for(int test_case = 1; test_case <= T; test_case++)
		{
			int N = sc.nextInt();
			int M = sc.nextInt();
			
			int[][] arr = new int[N][N];
			// 정답 리스트 선언
			List<Integer> answer_list = new ArrayList<>();
			// 전체 맵 입력 받기
			for (int r = 0; r < N; r++) {
				for (int c = 0; c < N; c++) {
					arr[r][c] = sc.nextInt();
				}
			}
			// 전체 순회하면서 찾기
			int answer;
			for (int r = 0; r < N-M+1; r++) {
				for (int c = 0; c< N-M+1; c++) {
					answer = 0;
					for (int i =0; i<M; i++) {
						for (int j = 0; j<M; j++) {
							answer += arr[r+i][c+j];
						}
					}
					answer_list.add(answer);
				}
			}
			
			int max = Collections.max(answer_list);
			System.out.println("#" + test_case + " " + max);
		}
	}
}